### PR TITLE
Add package for compressed tar files

### PR DESCRIPTION
**Pull request by jbrockopp**
Adding the functionality for compressed tar files using gzip. Created unit tests for the package. Manually tested by forking my own drone-s3-cache repo and made changes to the plugin file to use the tgz package instead of the default tar package and it worked properly.

### CODE REVIEW

**Comment by jmccann** [comment]
LGTM

@donny-dont Thoughts?

File: archive/tgz/tgz.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"io"
+	"compress/gzip"
+
+	. "github.com/drone/drone-cache-lib/archive"
+	"github.com/drone/drone-cache-lib/archive/tar"
+)
+
+type tgzArchive struct{}
+
+// NewTarArchive creates an Archive that uses the .tar file format.
```
Comment (donny-dont) [inline archive/tgz/tgz.go]: Comment is incorrect here.

**Review by donny-dont — CHANGES_REQUESTED** [review]
Fix that and good to merge

File: archive/tgz/tgz.go (hunk 1 above)
Comment (jbrockopp) [reply archive/tgz/tgz.go]: Comment has been fixed.

**Comment by jbrockopp** [comment]
Fixed comment. @donny-dont @jmccann

**Review by donny-dont — APPROVED** [review]

**Pull request merged**